### ISSUE TITLE
Outlook 2010: No Shortcut Key Given for the To Button in the New Message Window

### ISSUE THREAD

**Issue by elliott94** [nvaccess/nvda#10807]
### Steps to reproduce:

From within Outlook 2010, open a new message and shift + tab to the "To..." button

### Actual behavior:

When NVDA encounters the button, it outputs the following: "To... button alt +" - it is missing part of the shortcut key

### Expected behavior:

NVDA should either output the entirety of the shortcut key for this button (if one actually exists), or simply ignore the "alt +" suffix for this particular button

**Comment by lukaszgo1** [comment]
It took me a moment to realize what is going on there, but assuming that it behaves in the same way in Polish and English versions of Outlook the shortcut for this button is alt+period. Therefore on punctuation levels on which period is not spoken the shortcut seems to be incomplete. @elliott94 Can you confirm?

**Comment by elliott94** [comment]
Ah - this is exactly it, thank you for clarifying. :) Could we possibly consider explicitly announcing the period character in this case, or would this be out-of-scope?

**Comment by lukaszgo1** [comment]
While I do not think special casing of this particular button is a good idea, forcing shortcus to be read at maximum punctuation seems reasonable to me.

**Comment by CyrilleB79** [comment]
In French the "To..." button is labelled 'À...' and the shortcut is 'Alt+à'. However, it is announced 'Alt+a' instead of 'Alt+a grave' because 'à' is also a full word in French.

Using the character punctuation level to pronounce the part after the (last) '+' would probably resolve this problem.

**Comment by DrSooom** [comment]
But please not the plus sign alone. Adding an option for common hotkey combos only like ALT+, SHIFT+, Shift+, CTRL+ (they all must be localized) should handle this in a more desired way. Otherwise: This sentence ends with LGPL 2.1+. And then the full stop after the plus sign would be spoken too.

**Comment by Adriani90** [comment]
I think actually forcing NVDA to report punctuation for shortcuts in any case it is a reasonable idea, even if there are two + signs in the shortcut. The way how the punctuation of the shortcut is translated however, can be adjusted in the symbols.dic on the translation system and is out of scope for this particular issue.
cc: @michaelDCurran

**Comment by DrSooom** [comment]
Localization is required to detect possible hotkeys which a visible in menus or even in plain text. We aren't talking about the ALT key shortcuts alone.

**Comment by Adriani90** [comment]
This has to be performed in the gestures.ini and this unfortunately is another issue because gestures.ini is not automatically integrated into the automatic translation workflow. Thus, this has to be done manually by the community and I would say it still remains out of scope for this issue.

 

 

 

Von: Daniel Mayr <notifications@github.com> 
Gesendet: Dienstag, 25. Februar 2020 21:54
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Comment <comment@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Outlook 2010: No Shortcut Key Given for the To Button in the New Message Window (#10807)

 

Localization is required to detect possible hotkeys which a visible in menus or even in plain text. We aren't talking about the ALT key shortcuts alone.

**Comment by DrSooom** [comment]
@Adriani90: I guess we are talking about two fully different things. I'm talking about how NVDA is able to detect that WIN+E represents a shortcut. And if this is the case, NVDA should speak "Win plus E" instead of "Win E", even if the plus sign isn't spoken in the current selected symbol level – by giving the end user a new checkbox to enable this function. Same with Win++ or Win+. where the last Unicode character has to be spoken independently from the current selected symbol level. And here all possible shortcut prefixes must be translated for that detection, otherwise only English shortcuts would be detected correctly.
